### PR TITLE
Add All Abstract-Inheriting Tests to E2E Script

### LINKED ISSUE #557: Check that all E2E Tests are Complete and Running
The E2E Tests are all abstract, but at the bottom of the file inherit multiple contracts and run as non-abstract. The last merge with HAI may not have considered incomplete E2E test contracts.

Objective:
- check that are abstract contracts are being used
- check that all E2E tests are running in `yarn test`

### PR DESCRIPTION
**Pull request by daopunk**
closes #557

### CODE REVIEW

File: test/e2e/E2EStabilityFeeTreasury.t.sol [hunk 1]
```diff
@@ -31,7 +31,7 @@ abstract contract E2EStabilityFeeTreasuryTest is BaseUser, Common {
     // giving 25% of the balance to bob
     uint256 _rad = _coinBalance / 4;
 
-    vm.prank(governor);
+    vm.prank(deployer);
```
Comment (pi0neerpat) [inline test/e2e/E2EStabilityFeeTreasury.t.sol:34]: Why is deployer being swapped with governor in this file?

File: test/e2e/E2ETest.t.sol [hunk 2]
```diff
@@ -135,7 +151,33 @@ abstract contract E2ETest is BaseUser, Base_CType, Common {
...
     assertEq(_expectedCollateral, COLLATERAL_AMOUNT);
 
-    _joinCoins(address(this), _amountToBid);
+    _buyCollateral(address(this), address(collateralAuctionHouse[_cType()]), 1, _expectedCollateral, _amountToBid);
+
+    uint256 _decimals = collateral[_cType()].decimals();
+    uint256 _collateralWei = _expectedCollateral / 10 ** (18 - _decimals);
+    assertEq(collateral[_cType()].balanceOf(address(this)) - _initialBalance, _collateralWei);
+
+    // NOTE: auctions(1) is deleted
```
Comment (pi0neerpat) [inline test/e2e/E2ETest.t.sol:160]: Do we need to create an issue to continue watching this?

**Review by pi0neerpat — CHANGES_REQUESTED** [review]
Due to the number of changes, lets review this together

File: test/e2e/E2EStabilityFeeTreasury.t.sol (hunk 1 above)
Comment (daopunk) [reply test/e2e/E2EStabilityFeeTreasury.t.sol:34]: that was a HAI change. previously they used a fake governor for testing, but i guess decided to just use the deployer instead

File: test/e2e/E2ETest.t.sol (hunk 2 above)
Comment (daopunk) [reply test/e2e/E2ETest.t.sol:160]: watch what specifically?

**Review by pi0neerpat — APPROVED** [review]

**Pull request merged**